### DISCUSSION TITLE
How to handle error with useSWRMutation?

### DISCUSSION THREAD

**Discussion by utg1-tdawg** [vercel/swr#2450] [Q&A]
The error handler that I pass to `onError` in `useSWRMutation` gets called when an error occurs. However, it seems like the error doesn't get caught which results in an unhandled runtime error. 

Here is my code:
```typescript
function useCreatePost() {
  return useSWRMutation(
    "POST",
    () =>
      // this url is fake and should return a http error with status code 404
      axios.post("https://jsonplaceholder.typicode.com/FAKE", {
        title: "foo",
        body: "bar",
        userId: 1,
      }),
    {
      onError: () => console.log("Mutation error"),
    }
  );
}

export default function Post() {
  const { trigger } = useCreatePost();

  return <button onClick={trigger}>Create post</button>;
}
```

The unhandled runtime error:
```
Uncaught (in promise) 
AxiosError {message: 'Request failed with status code 404', name: 'AxiosError', code: 'ERR_BAD_REQUEST', config: {…}, request: XMLHttpRequest, …}
```

Any help will be greatly appreciated! 🙏

**Comment by koba04** [comment]
`trigger` throws an error regardless of `onError`. You can make it disable by `throwOnError = false`.

https://swr.vercel.app/docs/mutation#parameters-1

**Reply by utg1-tdawg** [reply]
Thanks. But why does `trigger` throw an `error` by default? Doesn't that negate the purpose of `onError` which is supposed to be an error handler? Would just like to know if I am missing something before sprinkling `throwOnError = false` across all mutation hooks.
